java-to-yup-core/src/main/java/sh/ivan/yup/PropertyDescriptor.java

```diff
@@ -5,30 +5,33 @@
 import java.lang.reflect.AnnotatedElement;
 import java.lang.reflect.Field;
 import java.lang.reflect.Method;
+import java.lang.reflect.Type;
 import java.util.HashSet;
 import java.util.Objects;
+import java.util.Optional;
 import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import lombok.Data;
 
 @Data
 public class PropertyDescriptor {
-    private final PropertyModel propertyModel;
-    private final Field field;
+    private final Type type;
     private final Set<AnnotatedElement> annotatedElements;
```

Comment:
This should probably be renamed to something like TypeDescriptor: it no longer necessarily corresponds to an object's property, but represents a typed element (a property, an array property's component type, or even a top level object's type)